lib/ide/src/view/ui_component.rs

```diff
@@ -0,0 +1,52 @@
+//! This module contains the base definition of UiComponent.
+
+use nalgebra::Vector2;
+
+
+
+// ===============
+// === Padding ===
+// ===============
+
+/// A struct containing the padding values.
+#[derive(Clone,Copy,Debug,Default)]
+pub struct Padding {
+    /// Left padding.
+    pub left   : f32,
+
+    /// Top padding.
+    pub top    : f32,
+
+    /// Right padding.
+    pub right  : f32,
+
+    /// Bottom padding;
+    pub bottom : f32
+}
+
+
+
+// ===================
+// === UiComponent ===
+// ===================
+
+/// A trait defining common interfaces of UI components.
+pub trait UiComponent {
```

Comment:
As we agreed, please rename it @notdanilo .

Reply:
I renamed this trait to `TemporaryPanel`, the file is now called `temporary_panel.rs` and `Padding` is now called `TemporaryPadding`. The module documentation was updated accordingly.